go/libraries/doltcore/sqle/enginetest/dolt_server_test.go

```diff
@@ -336,3 +347,92 @@ func newConnection(t *testing.T, serverConfig sqlserver.ServerConfig) (*dbr.Conn
 	sess := conn.NewSession(nil)
 	return conn, sess
 }
+
+func TestDoltServerRunningUnixSocket(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("we define socket flag and unix socket cannot be created on Windows, so it will return error")
+	}
+
+	// TODO : socket flag is defined in this startServer method for now
+	unixSocketPath := "/tmp/mysql.sock"
+	// Running unix socket server
+	sc, serverConfig := startServer(t, false, "", unixSocketPath)
+	sc.WaitForStart()
+	require.True(t, strings.Contains(sqlserver.ConnectionString(serverConfig, "dolt"), "unix"))
+
+	t.Run("connecting mysql works", func(t *testing.T) {
+		conn, sess := newConnection(t, serverConfig)
+		// Assertions
+		rows, err := sess.Query("select 1")
+		require.NoError(t, err)
+		assertResultsEqual(t, []sql.Row{{1}}, rows)
+		require.NoError(t, conn.Close())
+
+	})
+
+	t.Run("connecting to local server with both tcp and socket connections", func(t *testing.T) {
+		// default unix socket connection
+		localConn, localSess := newConnection(t, serverConfig)
+		rows, err := localSess.Query("select 1")
+		require.NoError(t, err)
+		assertResultsEqual(t, []sql.Row{{1}}, rows)
+
+		// connect with port defined
+		serverConfigWithPortOnly := sqlserver.DefaultServerConfig().WithPort(3306)
+		conn1, sess1 := newConnection(t, serverConfigWithPortOnly)
+		rows1, err := sess1.Query("select 1")
+		require.NoError(t, err)
+		assertResultsEqual(t, []sql.Row{{1}}, rows1)
+
+		// connect with host defined
+		serverConfigWithPortandHost := sqlserver.DefaultServerConfig().WithHost("127.0.0.1")
+		conn2, sess2 := newConnection(t, serverConfigWithPortandHost)
+		rows2, err := sess2.Query("select 1")
+		require.NoError(t, err)
+		assertResultsEqual(t, []sql.Row{{1}}, rows2)
+
+		// connect with port and host defined
+		serverConfigWithPortandHost1 := sqlserver.DefaultServerConfig().WithPort(3306).WithHost("0.0.0.0")
+		conn3, sess3 := newConnection(t, serverConfigWithPortandHost1)
+		rows3, err := sess3.Query("select 1")
+		require.NoError(t, err)
+		assertResultsEqual(t, []sql.Row{{1}}, rows3)
+
+		// close connections
+		require.NoError(t, conn3.Close())
+		require.NoError(t, conn2.Close())
+		require.NoError(t, conn1.Close())
+		require.NoError(t, localConn.Close())
+	})
+
+	// Stopping unix socket server
+	sc.StopServer()
+	err := sc.WaitForClose()
+	require.NoError(t, err)
+
+	// Running TCP socket server
+	tcpSc, tcpServerConfig := startServer(t, true, "0.0.0.0", "")
+	tcpSc.WaitForStart()
+	require.False(t, strings.Contains(sqlserver.ConnectionString(tcpServerConfig, "dolt"), "unix"))
+
+	t.Run("host and port specified, there should not be unix socket created", func(t *testing.T) {
+		// unix socket connection should fail
+		localServerConfig := sqlserver.DefaultServerConfig()
```

Comment:
Wouldn't `DefaultServerConfig()` return a config with unix socket set to "", so the connection on the next line wouldn't try to use a unix socket? Let me know if I'm just missing something obvious...